### PR TITLE
fix #1489, remove terminated executor from ExecutorManager

### LINKED ISSUE #1489: terminated executor is not removed from ExecutorManager

### PR DESCRIPTION
**Pull request by manuzhang**
[Review on Reviewable](https://reviewable.io/reviews/gearpump/gearpump/1490)

### CODE REVIEW

**Comment by clockfly** [comment]
+1

**Comment by stanleyxu2005** [comment]
The patch will help to hide executors when related processor is terminated, right?
Can I reference it to https://github.com/gearpump/gearpump/issues/1401

**Comment by clockfly** [comment]
There is slight difference.

Here actually means the executor is really dead. But in #1401, the executor still alive, but hidden because it will not handle any message in future due to its lifetime definition.

File: streaming/src/main/scala/io/gearpump/streaming/appmaster/ExecutorManager.scala [hunk 1]
```diff
@@ -121,9 +121,12 @@ private[appmaster] class ExecutorManager(
   def terminationWatch : Receive = {
     case Terminated(actor) =>
       val executorId = Try(actor.path.name.toInt)
-      LOG.error(s"Executor $executorId is down")
       executorId match {
-        case scala.util.Success(id) => taskManager ! ExecutorStopped(id)
+        case scala.util.Success(id) => {
+          executors -= id
+          LOG.error(s"Executor $id is down")
```
Comment (stanleyxu2005) [inline streaming/src/main/scala/io/gearpump/streaming/appmaster/ExecutorManager.scala]: Not big thing, but I'd like to get clear. `LOG.error` means the executor is terminated **unexpectedly**. Expected behavior should be logged with `INFO`.

Comment (manuzhang) [inline streaming/src/main/scala/io/gearpump/streaming/appmaster/ExecutorManager.scala]: yes. it means the executor is terminated unexpectedly

**Pull request merged**